content/series/101/2_development_environment/index.md

```diff
@@ -0,0 +1,150 @@
+
+```cmake
+# set the json path to a variable, it will be needed multiple times
+set(CCJSON "${CMAKE_CURRENT_BINARY_DIR}/compile_commands.json")
+
+# check if the file exists
+if(EXISTS "${CCJSON}")
+    # log a status message to confirm it works
+	message(STATUS "Copying ${CCJSON} to project root")
+
+    # copy the file
+	file(COPY "${CCJSON}" DESTINATION "${PROJECT_SOURCE_DIR}")
+endif()
+```
+
+Check clangd's logs in VSCode's **Output** window to confirm that it isn't reporting any errors.
+
+> _**Note:** CMake, `compile_commands.json`, and clangd don't work well together when using **precompiled headers**; in such cases create a dummy config as a copy of `Debug` but with PCH off, and copy and use the JSON generated there._
+
+#### clang-format
+
+Most C++ programmers spend a considerable amount of time meticulously formatting their precious code, until they make an acquaintance with an autoformatter - of which `clang-format` is the indisputed champion - after which they pretty much never hand-format code again. This is the secret behind the reams of impossibly perfectly formatted and documented library code, for example throughout the standard library headers.
+
+It works by way of a configuration file in the root, much like `compile_commands.json`, called `.clang-format`, which the `clang-format` binary reads for the configured [**format style options**](https://clang.llvm.org/docs/ClangFormatStyleOptions.html). To start using it, create a `.clang-format` in the project root and select "Format document with clangd" through VSCode's command palette. VSCode also offers an option to auto-format documents on save (using the configured formatter).
+
+#### Compiler Warnings and clang-tidy
+
+Since C++ standards are generally very meticulous about backwards compatibility, not-so-great language and library design decisions made in the past cannot easily be rectified through breaking changes, and older programs must not suddenly stop compiling with a new compiler (provided they weren't engaging in unspecified behaviour in the first place). Ergo, there are a few language potholes and traps where the compiler can only warn you - cannot issue an error and stop (unless you configure it to be that way for a specific set of / all warnings), and which is why modern C++ development is greatly aided by turning up compiler warnings and using linters that perform static analysis outside compilation (perhaps even risking some false positive warnings). `clang++` offers a `g++`-like front-end, and accepts all those compiler flags (and many more); at the bare minimum it is highly recommended to set at least the following, for **every single C++ project you author**:
```

Comment:
> Ergo, there are a few language potholes and traps where the compiler can only warn you - cannot issue an error and stop (unless you configure it to be that way for a specific set of / all warnings), and which is why modern C++ development is greatly aided by turning up compiler warnings and using linters that perform static analysis outside compilation (perhaps even risking some false positive warnings).

This is a very long sentence. Your text is more accessible for non-native speakers and younger readers if you break it up a little. For example:

_Ergo, there are a few language potholes and traps where the compiler can only issue warnings for you unless you configure the compiler to interpret (all) warnings as errors. Modern C++ development is greatly aided by turning up compiler warnings and using linters that perform static analysis outside compilation, though this practice sometimes risks false positives._

> [...] to set at least the following __options__ for every [...]